File: https:/github.com/kuldeeppatel-cg/Leetcode.git/9-PalindromeNumber/9-PalindromeNumber.js
// Last updated: 17/07/2026, 14:20:36
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
     let reverse = 0;
   let num = x;
    while(num>0){
       let dig = num%10;
       reverse = reverse*10 + dig;
       num = Math.floor(num/10);
    }
    let check = (reverse == x)?true:false;
    return check;
};

File: https:/github.com/kuldeeppatel-cg/Leetcode.git/9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 17/07/2026, 14:20:34
class Solution {
public:
    bool isPalindrome(int x) {
        string s = to_string(x);
        if(x < 0){
            return false;
        }
        int i = 0;
        int j = s.length() - 1;
        while (j > i) {
            if (s[i] == s[j]) {
                i++;
                j--;
            } else {
                return false;
            }
        }
        return true;
    }
};